BepInEx.IL2CPP/Preloader.cs

```diff
@@ -83,5 +86,48 @@ public static void Run()
                 throw;
             }
         }
+
+        private static void InitializeUnityVersion()
+        {
+            try
+            {
+                ConfigEntry<String> config = ConfigFile.CoreConfig.Bind(
+                    "Unity", "Version",
+                    String.Empty,
+                    "Allows you to manually set the Unity version .");
+
+                String version = config.Value;
+                if (String.IsNullOrWhiteSpace(version))
+                {
+                    version = //Version.Parse(Application.unityVersion);
+                        Process.GetCurrentProcess().MainModule.FileVersionInfo.FileVersion;
+
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from main application module: [{version}]");
+                }
+                else
+                {
+                    PreloaderLogger.Log.LogDebug($"Unity version obtained from the config: [{version}]");
```

Comment:
Instead of Debug logging here I'd make a Warning level log and report that the Unity version is read from config. This will help to detect bad configuration.